### PR TITLE
Accessibility Spike

### PR DESCRIPTION
**Pull request by thestephenmarshall**
This PR adds console, Jest and Rspec Accessibility test capabilities. The console test will automatically run against any kit example in all views. 

#### Screens

accessibility 2021-02-04 12_10_52
Screen Shot 2021-02-04 at 12 10 02 PM
Screen Shot 2021-02-04 at 12 09 26 PM

#### Breaking Changes

No

#### Runway Ticket URL

https://nitro.powerhrg.com/runway/backlog_items/NUXE-451

#### How to test this

1. Simply go to a kit example and use the inspector
2. Run the jest test individually (where accessibility test is implemented) and view the result
3. Run the spec test individually (where accessibility test is implemented) and view the result

#### Checklist:

- [x] **LABELS** Add a label: `enhancement`, `bug`, `improvement`, `new kit`, `deprecated`, or `breaking`. See [Changelog & Labels](https://github.com/powerhome/playbook/wiki/Changelog-&-Labels) for details.
- [x] **URGENCY** Please select a release milestone
- [x] **DEPLOY** Please add the `Milano` label when you are ready for a review.
- [x] **SCREENSHOT** Please add a screen shot or two.
- [x] **SPECS** Please cover your changes with specs.
- [x] **READ DOCS** Please make sure you have read and understand the [Playbook Release Process](https://github.com/powerhome/playbook/wiki/Playbook-Releases)

*The normal release cut off deadline is 3p EDT each week. Please reach out to the release team if you have an urgent request or need a release off cycle.*

### CODE REVIEW

File: playbook/app/pb_kits/playbook/pb_image/_image.jsx [hunk 1]
```diff
@@ -30,7 +30,9 @@ const Image = (props: ImageProps) => {
 
   const ariaProps = buildAriaProps(aria)
   const classes = classnames(
-    buildCss('pb_image_kit lazyload blur_up', size),
+    buildCss('pb_image_kit', size),
+    'lazyload',
+    'blur_up',
```
Comment (thestephenmarshall) [inline playbook/app/pb_kits/playbook/pb_image/_image.jsx:35]: These were out of order and now the Rails and React generated class names are consistent.

File: playbook/app/pb_kits/playbook/pb_image/image.rb [hunk 2]
```diff
@@ -16,7 +16,7 @@ class Image
       prop :url
 
       def classname
-        generate_classname("pb_image_kit lazyload blur_up", size_class) + rounded_class
+        generate_classname("pb_image_kit#{size_class} lazyload blur_up") + rounded_class
```
Comment (thestephenmarshall) [inline playbook/app/pb_kits/playbook/pb_image/image.rb:19]: Just a little re-ordering here solved the issue with classnames being out of order and generally wrong.

File: playbook/app/pb_kits/playbook/utilities/accessibility.js [hunk 3]
```diff
@@ -0,0 +1,22 @@
...
+        console.warn('🚨 [axe-core] Accessibility issues found. See below for a list:')
+        console.dir(results.violations)
+      } else {
+        console.log('[axe-core] Yay! 🎉 No accessibility violations were found!')
+      }
+    })
+    .catch((err) => {
+      console.error('Something bad happened:', err.message)
+    })
+}
```
Comment (thestephenmarshall) [inline playbook/app/pb_kits/playbook/utilities/accessibility.js:22]: This is the console accessibility check in the browser. It runs only in development mode, always.

File: playbook/lib/generators/kit/templates/kit_jsx_test.erb.tt [hunk 4]
```diff
@@ -8,14 +8,10 @@ import { <%= @kit_name_pascal %> } from '../'
...
-    />
-  )
+  const props = {
+    data: { testid: 'default' }
+  }
 
-  const kit = screen.getByTestId(testId)
+  const kit = renderKit(<%= @kit_name_pascal %> , props)
   expect(kit).toBeInTheDocument()
 })
```
Comment (thestephenmarshall) [inline playbook/lib/generators/kit/templates/kit_jsx_test.erb.tt:17]: New kits have the updated test pattern.

File: playbook/spec/pb_kits/playbook/kits/image_spec.rb [hunk 5]
```diff
@@ -15,7 +15,7 @@
   describe "#classname" do
     it "returns namespaced class name", :aggregate_failures do
       expect(subject.new({}).classname).to eq "pb_image_kit lazyload blur_up"
-      expect(subject.new({size: "xs"}).classname).to eq "pb_image_kit lazyload blur_up_xs"
+      expect(subject.new({size: "xs"}).classname).to eq "pb_image_kit_xs lazyload blur_up"
```
Comment (thestephenmarshall) [inline playbook/spec/pb_kits/playbook/kits/image_spec.rb:18]: Here you can see the classname inconsistency.

File: playbook/app/pb_kits/playbook/utilities/accessibility.js [hunk 6]
```diff
@@ -0,0 +1,22 @@
+/* eslint-disable no-console */
+
+import axe from 'axe-core'
+
+export default function runAxe(include = '.pb--kit-example', ignore = ['.pb--kit-example :first-child']) {
```
Comment (coleerikson) [inline playbook/app/pb_kits/playbook/utilities/accessibility.js:5]: Is the space between example and : intended? `['.pb--kit-example :first-child']`

Comment (thestephenmarshall) [reply playbook/app/pb_kits/playbook/utilities/accessibility.js:5]: Yep. This is the only way I could get Axe to ignore those captions inside each example.

**Review by coleerikson — APPROVED** [review]
One quick question in the comments.

**Review by jasoncypret — APPROVED** [review]

**Pull request merged**